### PR TITLE
Add connection interface timeout config and remove Ctrl+G timer

### PR DESCRIPTION
**Pull request by AMZN-puvvadar**
This change adds per connection interface configuration of whether connection timeout should be enforced. Enforcement requires an opt-in from both the connection interface and the global cvars. The connection interface is opt-in by default.

Additionally, defaulting the Editor Connection Interface used to drive the Ctrl+G UX to not enforcing timeout which eliminates the need for the 15s timer. This connection cleans itself up once its job is done as well.

Signed-off-by: puvvadar <puvvadar@amazon.com>

### CODE REVIEW

File: Code/Framework/AzNetworking/AzNetworking/Framework/INetworkInterface.h [hunk 1]
```diff
@@ -103,6 +103,14 @@ namespace AzNetworking
         //! @return boolean true on success
         virtual bool Disconnect(ConnectionId connectionId, DisconnectReason reason) = 0;
 
+        //! Sets whether this connection interface can disconnect by virtue of a timeout
+        //! @param doesTimeout If this connection interface will automatically disconnect due to a timeout
+        virtual void SetDoesTimeout(bool doesTimeout) = 0;
```
Comment (kberg-amzn) [inline Code/Framework/AzNetworking/AzNetworking/Framework/INetworkInterface.h:108]: Possibly SetTimeoutEnabled(bool) and IsTimeoutEnabled() works better than Does?

**Review by kberg-amzn — APPROVED** [review]
Looks good!

**Review by AMZN-Gene — APPROVED** [review]

**Review by AMZN-Olex — APPROVED** [review]
How is the experience looking with this change? Could we use some UI text or even a log message if there is some down time before the connection in the Editor is made to the server?

File: Code/Framework/AzNetworking/AzNetworking/Framework/INetworkInterface.h [hunk 2]
```diff
@@ -103,6 +103,14 @@ namespace AzNetworking
         //! @return boolean true on success
         virtual bool Disconnect(ConnectionId connectionId, DisconnectReason reason) = 0;
 
+        //! Sets whether this connection interface can disconnect by virtue of a timeout
+        //! @param timeoutEnabled If this connection interface will automatically disconnect due to a timeout
+        virtual void SetTimeoutEnabled(bool timeoutEnabled) = 0;
+
+        //! Whether this connection interface will disconnect by virtue of a time out (does not account for cvars affecting all connections)
+        //! @return boolean true if this connection will not disconnect on timeout (does not account for cvars affecting all connections)
+        virtual bool IsTimeoutEnabled() = 0;
```
Comment (SergeyAMZN) [inline Code/Framework/AzNetworking/AzNetworking/Framework/INetworkInterface.h:112]: seems the other getters are const..
should this one be too?

**Review by SergeyAMZN — APPROVED** [review]

File: Code/Framework/AzNetworking/AzNetworking/Framework/INetworkInterface.h (hunk 2 above)
Comment (AMZN-puvvadar) [reply Code/Framework/AzNetworking/AzNetworking/Framework/INetworkInterface.h:112]: Ah good point, will update

**Comment by AMZN-puvvadar** [comment]
> How is the experience looking with this change? Could we use some UI text or even a log message if there is some down time before the connection in the Editor is made to the server?

Pretty similar honestly. The server launches immediately but takes a few seconds to stand itself up and start receiving traffic. The time gap is just minimized now and the Editor is no longer locked up during said gap.

**Pull request merged**